### PR TITLE
TST: Adding to_dict numeric consistency test (#22620)

### LINKED ISSUE #22620: DataFrame.to_dict(orient='records') numeric inconsistency
#### Code Sample, a copy-pastable example if possible

Similar to #11247, but related to numerics. If there are any object columns in the DataFrame, then numerical columns are turned into native types, whereas if all columns are numpy numerics, they stay that way:

```python
import pandas as pd

df = pd.DataFrame({
	'a': [1, 2, 3],
	'b': [1.0, 2.0, 3.0],
	'c': ['X', 'Y', 'Z']
})

dict_with_obj = df.to_dict('records')
print(f"Output types when an object column is present:")
print((type(dict_with_obj[0]['a']), type(dict_with_obj[0]['b'])))

dict_no_obj = df.drop(columns=['c']).to_dict('records')
print(f"Output types when no object column is present:")
print((type(dict_no_obj[0]['a']), type(dict_no_obj[0]['b'])))
```

Output:
```
Output types when an object column is present:
(<class 'int'>, <class 'float'>)
Output types when no object column is present:
(<class 'numpy.float64'>, <class 'numpy.float64'>)
```

#### Problem description

I need to combine the output dictionary with other dictionaries, and then convert it to JSON (using `ujson.dumps`). Since `ujson` cannot handle NumPy types, it crashes. Since I need to post process the output of `to_dict` before converting to JSON unfortunately `df.to_json` is not an option.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.0.final.0
python-bits: 64
OS: Darwin
OS-release: 17.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

pandas: 0.23.3
pytest: None
pip: 18.0
setuptools: 40.0.0
Cython: 0.28.5
numpy: 1.15.0
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by sofiane87**
- [x] closes #22620
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/frame/test_convert_to.py [hunk 1]
```diff
@@ -612,3 +612,19 @@ def test_to_dict_wide(self):
...
+
+    def test_to_dict_orient_dtype(self):
+        # https://github.com/pandas-dev/pandas/issues/22620
+        # Input Data
+        input_data = {"a": [1, 2, 3], "b": [1.0, 2.0, 3.0], "c": ["X", "Y", "Z"]}
+        df = DataFrame(input_data)
+        # Expected Dtypes
+        expected = {"a": int, "b": float, "c": str}
+        # Extracting dtypes out of to_dict operation
+        df_dict = df.to_dict("records")[0]
```
Comment (jreback) [inline pandas/tests/frame/test_convert_to.py]: can you compare the entire result, not just the first record

Comment (sofiane87) [reply pandas/tests/frame/test_convert_to.py]: this is now done

File: pandas/tests/frame/test_convert_to.py [hunk 2]
```diff
@@ -612,3 +612,19 @@ def test_to_dict_wide(self):
...
+        # Expected Dtypes
+        expected = {"a": int, "b": float, "c": str}
+        # Extracting dtypes out of to_dict operation
+        for df_dict in df.to_dict("records"):
+            result = {
+                "a": type(df_dict["a"]),
+                "b": type(df_dict["b"]),
+                "c": type(df_dict["c"]),
+            }
+            assert result == expected
```
Comment (gfyoung) [inline pandas/tests/frame/test_convert_to.py]: I would check that the result of `df.dtypes` is correct.  That would be simpler than this for-loop.

Comment (sofiane87) [reply pandas/tests/frame/test_convert_to.py]: the point is to check that the `type` returned from `to_dict` are the same as the ones within the dataset. I am not sure how to do that using `df.dtypes`. @simonjayhawkins recommended that I make the tests as explicit as possible.

Comment (jreback) [reply pandas/tests/frame/test_convert_to.py]: actually this is ok

Comment (jreback) [reply pandas/tests/frame/test_convert_to.py]: yeah this is enough here

**Pull request merged**